packages/miniflare/src/plugins/core/index.ts

```diff
@@ -267,9 +270,13 @@ function getCustomServiceDesignator(
 			serviceName = getBuiltinServiceName(workerIndex, kind, name);
 		}
 	} else if (service === kCurrentWorker) {
-		// Sets SELF binding to point to router worker instead if assets are present.
+		// Sets SELF binding to point to the Router Worker or the Assets
+		// Proxy Worker (depending on whether `unsafeEnableAssetsRpc` is set)
+		// if assets are present.
 		serviceName = hasAssetsAndIsVitest
-			? `${ROUTER_SERVICE_NAME}:${refererName}`
+			? unsafeEnableAssetsRpc
```

Comment:
supporting this for vitest-pool-workers made things more complicated than I would have wanted, because it required introducing `unsafeEnableAssetsRpc` on the vitest pool workers side as well. Does it make sense to support this experimental feat in pool workers at the moment?

Reply:
that will actually be needed since `unsafeEnableAssetsRpc` is only ever set if `--x-assets-rpc` is passed to `wrangler dev`. Therefore, vitest-pool-workers needs a similar config key, smth like 👇 , to be able to correctly execute this conditional logic. Otherwise where would the `unsafeEnableAssetsRpc` value ever came from, if not in a `wrangler dev` context?

```
test: {
    poolOptions: {
      workers: {
        miniflare: {
          unsafeEnableAssetsRpc: true
        }
      },
    },
  },
  ```